core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -455,7 +455,9 @@ class LogManager(logDirs: Array[File],
         }
       }
     }
-    checkpointLogRecoveryOffsets()
+
+    if (writeCheckpoint)
```

Comment:
Do we need to optimize that? With KIP-101, most of time there should be no log truncation when a replica becomes the follower. Perhaps we could return a boolean in log.truncateTo() to indicate whether a truncation happens or not and only call checkpointLogRecoveryOffsets() if the truncation happens. This way, we probably can get rid of writeCheckpoint w/o losing much performance.

Reply:
Thanks for the review. Yeah good point. I have updated the patch as suggested.